onchain/rollups/contracts/library/LibProof.sol

```diff
@@ -10,7 +10,7 @@ library LibProof {
         Proof calldata proof
     ) internal pure returns (uint256 inputIndex) {
         inputIndex =
-            proof.inputRange.firstIndex +
+            uint256(proof.inputRange.firstIndex) +
```

Comment:
we can omit this cast

Reply:
If we omit the cast, it will do addition on `uint64` values, which might be prone to overflows.

Reply:
It only overflows if the input index is greater than 2^64, which will not happen right?

Reply:
Yes, one will not be able to prove anything with an input index greater than or equal to $2^{64}$, because there is no `InputRange` that would contain it. The key difference is that, from the perspective of the user, the error would be different.
If we do the addition in...
- `uint256`, the error would be `InputIndexOutOfRange`
- `uint64`, the error would be `Panic` with error code of an arithmetic overflow
I think the first is easier for the user to understand.